libdrgn/symbol.h

```diff
@@ -5,11 +5,20 @@
 #define DRGN_SYMBOL_H
 
 #include <stdint.h>
+#include <gelf.h>
+
+#include "drgn.h"
 
 struct drgn_symbol {
 	const char *name;
 	uint64_t address;
 	uint64_t size;
+	enum drgn_symbol_binding binding;
+	enum drgn_symbol_kind kind;
 };
 
+/** Populates the fields of @ref drgn_symbol. */
+void drgn_symbol_init_from_elf(const char *name, uint64_t address,
+			       GElf_Sym elf_sym, struct drgn_symbol *ret);
```

Comment:
`GElf_Sym` is 24 bytes, so I think we should pass it as a pointer rather than by value.

```suggestion
			       const GElf_Sym *elf_sym, struct drgn_symbol *ret);
```